modules/nwtc-library/src/VTK.f90

```diff
@@ -1,610 +1,614 @@
-                write(mvtk%vtk_unit,'(1'//RFMT//')')D
-            endif
-        endif
-    end subroutine
-    
-    
-    subroutine vtk_cell_data_vector(D,sname,mvtk)
-        real(ReKi), dimension(:,:),intent(in) :: D  !< 3 x n
-        character(len=*),intent(in) ::sname
-        type(FVW_VTK_Misc),intent(inout) :: mvtk
-        if ( mvtk%bFileOpen ) then
-            if (mvtk%bBinary) then
-                write(mvtk%vtk_unit)'VECTORS '//trim(sname)//' double'//NewLine
-                write(mvtk%vtk_unit)D
-                write(mvtk%vtk_unit)NewLine
-            else
-                write(mvtk%vtk_unit,'(A,A,A)') 'VECTORS ', sname, ' double'
-                write(mvtk%vtk_unit,'(3'//RFMT//')')D
-            endif
-        endif
-    end subroutine
-
-    ! --------------------------------------------------------------------------------}
-    ! --- VTK Tools 
-    ! --------------------------------------------------------------------------------{
-    !> Exports a Plane From a mesh
-    subroutine export_plane_grid3d(fname,v1,v2,v3,Values,mvtk)
-        character(len=*),intent(in)             :: fname
-        real(ReKi),dimension(:), intent(in)       :: v1,v2,v3
-        real(ReKi),dimension(:,:,:,:), intent(in) :: Values
-        type(FVW_VTK_Misc),intent(inout) :: mvtk
-        !  Variables
-        integer :: nD
-
-        ! Writting
-        if ( vtk_new_ascii_file(trim(fname),'grid',mvtk)) then
-            nD=size(Values,1)
-            call vtk_dataset_rectilinear(v1,v2,v3,mvtk)
-            ! Output as a structured grid, No need to reorder
-            call vtk_point_data_init(mvtk)
-            ! Could be a function of nDim, be careful
-            if(nD==3) then
-                call vtk_point_data_vector(Values(1:3,:,:,:),'Velocity',mvtk) ! Label...
-            endif
-
-            call vtk_close_file(mvtk)
-        endif ! file opening
-    end subroutine
-    
-    !> Exports a Plane From a mesh
-    subroutine export_plane_grid2d(fname,v1,v2,v3,Values,mvtk)
-        character(len=*),intent(in)             :: fname
-        real(ReKi),dimension(:), intent(in)       :: v1,v2,v3
-        real(ReKi),dimension(:,:,:), intent(in) :: Values
-        type(FVW_VTK_Misc),intent(inout) :: mvtk
-        !  Variables
-        integer :: nD
-
-        ! Writting
-        if ( vtk_new_ascii_file(trim(fname),'plane',mvtk) ) then
-            nD=size(Values,1)
-            call vtk_dataset_rectilinear(v1,v2,v3,mvtk)
-            ! Output as a structured grid, No need to reorder
-            call vtk_point_data_init(mvtk)
-            ! Could be a function of nDim, be careful
-            if(nD==3) then
-                call vtk_point_data_vector(Values(1:3,:,:),'Velocity',mvtk) ! Label...
-            endif
-
-            call vtk_close_file(mvtk)
-        endif ! file opening
-    end subroutine
-end module FVW_VTK
+module VTK
+    !use PrecisionMod, only: ReKi
+    use NWTC_Library, only: ReKi, GetNewUnit, NewLine, WrScr
```

Comment:
If VTK is in the library, does this become a circular dependency?

Reply:
That's a good point. So far I haven't put it in the "library module" explicitly, so it isn't a problem.